### PR TITLE
PR: Correctly handle programming language in calltips

### LINKED ISSUE #9194: Pass language from console and editor on calltip call
On spyder/widgets/mixins.py

```python
    _PARAMETER_HIGHLIGHT_COLOR = '#DAA520'
    _DEFAULT_TITLE_COLOR = '#2D62FF'
    _CHAR_HIGHLIGHT_COLOR = 'red'
    _DEFAULT_TEXT_COLOR = '#999999'
```

and 

```python
# TODO: Select language and pass it to call
```
----

@dalthviz we need to be able to ask the kernel what language it is running (right now python, but it could be anything)

spyder/plugins/ipythonconsole/widgets/help.py
```python
# TODO: Pass the language from the Console to the calltip
```

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #9194

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, I released qtconsole 4.5.0 a couple fo weeks ago, which your addition of `language_name`. So please add that requirement to our `setup.py` and mention it in our Readme.

After doing that, do you think this will be ready? Or do you plan to add more things here?

**Comment by dalthviz** [comment]
@ccordoba12 I left the PR as WIP since I was unsure if there was any other place to make changes. I will do the `setup.py` and `README` changes

File: spyder/widgets/mixins.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def handle_sub(matchobj):
 
         return title
 
-    def show_calltip(self, signature, parameter=None):
+    def show_calltip(self, signature, parameter=None, language="python"):
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: You probably need to pass `language` to the call to `_format_signature` inside this method.

But @goanpeca knows more about it. @goanpeca?

**Comment by ccordoba12** [comment]
@dalthviz, your changes broke calltips in the IPython console. Please take a look at that.

File: spyder/widgets/mixins.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ class BaseEditMixin(object):
     _DEFAULT_TITLE_COLOR = '#2D62FF'
     _CHAR_HIGHLIGHT_COLOR = 'red'
     _DEFAULT_TEXT_COLOR = '#999999'
-    _DEFAULT_LANGUAGE = 'python'
+    _DEFAULT_LANGUAGE = ''
```
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: I think we should use `'python'` here. Why did you decide to leave it empty?

Comment (dalthviz) [reply spyder/widgets/mixins.py]: I was checking the use of `getattr` for trying to get the `language` and the default values used are the empty `string` or `None`, should I revert the change and use `python` instead?

Comment (ccordoba12) [reply spyder/widgets/mixins.py]: Yeah, please do. I think it'd be easier to understand the code that way.

**Pull request merged**